sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ChangeFeedProcessorBuilder.java

```diff
@@ -127,17 +174,30 @@ public ChangeFeedProcessorBuilder options(ChangeFeedProcessorOptions changeFeedP
     public ChangeFeedProcessor buildChangeFeedProcessor() {
         validateChangeFeedProcessorBuilder();
 
-        ChangeFeedProcessorBuilderImpl builder = new ChangeFeedProcessorBuilderImpl()
-            .hostName(this.hostName)
-            .feedContainer(this.feedContainer)
-            .leaseContainer(this.leaseContainer)
-            .handleChanges(this.partitionKeyBasedLeaseConsumer);
-
-        if (this.changeFeedProcessorOptions != null) {
-            builder.options(this.changeFeedProcessorOptions);
+        if (ChangeFeedMode.LATEST_VERSION.equals(changeFeedMode)) {
+            ChangeFeedProcessorBuilderImpl builder = new ChangeFeedProcessorBuilderImpl()
+                .hostName(this.hostName)
+                .feedContainer(this.feedContainer)
+                .leaseContainer(this.leaseContainer)
+                .handleChanges(this.partitionKeyBasedLeaseConsumer);
+
+            if (this.changeFeedProcessorOptions != null) {
+                builder.options(this.changeFeedProcessorOptions);
+            }
+
+            return builder.build();
+        } else {
+            com.azure.cosmos.implementation.changefeed.fullfidelity.ChangeFeedProcessorBuilderImpl builder =
```

Comment:
should we also remove the fullfidelity from the namespace?
com.azure.cosmos.implementation.changefeed.allVersions.ChangeFeedProcessorBuilderImpl

Reply:
Yes we should, however, not in this PR. Since it is only implementation details, we can update it later. I don't want to make this PR more difficult to review :)